src/Storages/MergeTree/MergeTreeBaseSelectProcessor.h

```diff
@@ -82,14 +103,36 @@ class MergeTreeBaseSelectProcessor : public SourceWithProgress
     /// This header is used for chunks from readFromPart().
     Block header_without_virtual_columns;
 
-    std::unique_ptr<MergeTreeReadTask> task;
-
     std::shared_ptr<UncompressedCache> owned_uncompressed_cache;
     std::shared_ptr<MarkCache> owned_mark_cache;
 
     using MergeTreeReaderPtr = std::unique_ptr<IMergeTreeReader>;
     MergeTreeReaderPtr reader;
     MergeTreeReaderPtr pre_reader;
+
+    MergeTreeReadTaskPtr task;
+
+    std::optional<ParallelReadingExtension> extension;
+    std::vector<size_t> average_mark_size_bytes;
```

Comment:
Why this is needed at all? Looks like we never read it

Reply:
Yes. Deleted.